src/quilt/portablemc_quilt/__init__.py

```diff
@@ -0,0 +1,164 @@
+                    "game_version": game_version
+                }, done=True)
+
+            ver = QuiltVersion(ctx, game_version, loader_version, prefix=ctx.ns.fabric_prefix)
+            ver.manifest = manifest
+            return ver
+
+        return old(ctx, version_id)
+
+    # Messages
+
+    pmc.messages.update({
+        "args.start.quilt_prefix": "Change the prefix of the version ID when starting with Quilt.",
+        "start.quilt.resolving_loader": "Resolving quilt loader for {game_version}...",
+        "start.quilt.resolving_loader_specific": "Resolving quilt loader {loader_version} for {game_version}...",
+        "start.quilt.error.invalid_format": "To launch quilt, use 'quilt:[<mc-version>[:<loader-version>]]'.",
+        f"start.quilt.error.{QuiltVersionNotFound.GAME_VERSION_NOT_FOUND}": "Game version {version} not found.",
+        f"start.quilt.error.{QuiltVersionNotFound.LOADER_VERSION_NOT_FOUND}": "Loader version {version} not found."
+    })
+
+
+class QuiltVersion(Version):
+
+    def __init__(self, context: Context, game_version: str, loader_version: Optional[str], *, prefix: str = "fabric"):
+
+        """
+        Construct a new quilt version, such version are specified by a game version and an optional loader-version,
+        if loader-version is not specified, the latest version is used and fetched when first calling `prepare_meta`.
+        """
+
+        # If the loader version is unknown, we temporarily use a version ID without
+        id_ = f"{prefix}-{game_version}" if loader_version is None else f"{prefix}-{game_version}-{loader_version}"
+
+        super().__init__(context, id_)
+
+        self.game_version = game_version
+        self.loader_version = loader_version
+
+    # The function 'prepare_meta' might throw 'QuiltVersionNotFound' either
+    # '_prepare_id' or from the inner calls to '_fetch_version_meta'.
+
+    def _prepare_id(self):
+        # If the loader version is unknown, the version's id is not fully defined,
+        # so we add the loader version to the id.
+        if self.loader_version is None:
+            self.loader_version = request_loader_version(self.game_version)
+            self.id += f"-{self.loader_version}"
+
+    def _validate_version_meta(self, version_id: str, version_dir: str, version_meta_file: str, version_meta: dict) -> bool:
+        if version_id == self.id:
+            # If the version is installed, it is always valid, because we don't have any metadata to check its validity.
+            return True
+        else:
+            return super()._validate_version_meta(version_id, version_dir, version_meta_file, version_meta)
+
+    def _fetch_version_meta(self, version_id: str, version_dir: str, version_meta_file: str) -> dict:
+        if version_id != self.id:
+            return super()._fetch_version_meta(version_id, version_dir, version_meta_file)
+        return request_version_loader_profile(self.game_version, self.loader_version)
```

Comment:
During test, I found that the "id" field inside versions' metadata is unchanged and is not coherent with the file's name. I'll also change that in the fabric add-on later on.
```python
meta = request_version_loader_profile(self.game_version, self.loader_version)
meta["id"] = self.id
return meta
```

Reply:
updated